### PR TITLE
WAS14wrap

### PR DESCRIPTION
**Pull request by was4**
Finish wrapping SVCXPRT per vc, lots of bug fixes for really old code....

### CODE REVIEW

File: ntirpc/rpc/tirpc_compat.h [hunk 1]
```diff
@@ -67,7 +67,6 @@
 #define svc_tp_create(a, b, c, d) svc_tp_ncreate(a, b, c, d)
 #define svc_tli_create(a, b, c, d, e) svc_tli_ncreate(a, b, c, d, e)
 #define svc_vc_create(a, b, c) svc_vc_ncreate(a, b, c)
-#define svc_vc_create2(a, b, c, d) svc_vc_ncreate2(a, b, c, d)
```
Comment (dang) [inline ntirpc/rpc/tirpc_compat.h:70]: Do we care about tirpc compat?  If so, can we map svc_vc_create2() into some useful call?  It looks like it maps to svc_vc_ncreatef()...

File: src/vc_generic.c [hunk 2]
```diff
@@ -101,6 +101,7 @@ clnt_read_vc(XDR *xdrs, void *ctp, void *buf, int len)
 	return (len);
 }
 
+#if 0
```
Comment (dang) [inline src/vc_generic.c:104]: Is this coming back later?  If not, can you just nuke it?

File: src/vc_generic.c [hunk 3]
```diff
@@ -191,6 +190,7 @@ svc_read_vc(XDR *xdrs, void *ctp, void *buf, int len)
 	return (-1);
 }
 
+#if 0
```
Comment (dang) [inline src/vc_generic.c:193]: Ditto this.

File: src/vc_generic.c [hunk 4]
```diff
@@ -241,24 +241,18 @@ svc_write_vc(XDR *xdrs, void *ctp, void *buf, int len)
...
 		return (svc_read_vc(xdrs, ctp, buf, len));
-		break;
-	default:
-		/* better not */
-		abort();
-	}
+	return (clnt_read_vc(xdrs, ctp, buf, len));
 }
 
+#if 0
```
Comment (dang) [inline src/vc_generic.c:255]: And this

File: src/rpc_ctx.c [hunk 5]
```diff
@@ -74,139 +74,137 @@ rpc_msg_init(struct rpc_msg *msg)
...
 	CLIENT *clnt = ctx->ctx_u.clnt.clnt;
-	struct svc_vc_xprt *xd = (struct svc_vc_xprt *)clnt->cl_p1;
-	struct rpc_dplx_rec *rec = xd->rec;
+	struct cx_data *cx = CX_DATA(clnt);
+	struct rpc_dplx_rec *rec = cx->cx_rec;
+	struct svc_vc_xprt *xd = VC_DR(rec);
 
-	/* this lock protects both xid and rbtree */
-	REC_LOCK(rec);
+	/* the lock protects both xid and rbtree */
```
Comment (dang) [inline src/rpc_ctx.c:130]: You took out the lock, but left the comment.  The comment should probably be removed as well, or someone's going to think there's a missing lock.

File: src/svc_xprt.c [hunk 6]
```diff
@@ -102,14 +103,16 @@ void svc_xprt_init()
...
-		do { \
-			if (!initialized) \
-				svc_xprt_init(); \
-		} while (0); \
-	}
+static inline bool
+svc_xprt_init_failure(void)
+{
+	if (initialized)
+		return (false);
```
Comment (dang) [inline src/svc_xprt.c:113]: Should this be failing if already initialized?

**Review by dang — COMMENTED** [review]
Just a few nit questions. Looks good over all.

File: ntirpc/rpc/tirpc_compat.h (hunk 1 above)
Comment (was4) [reply ntirpc/rpc/tirpc_compat.h:70]: There is no tirpc svc_vc_create2() -- that is entirely ntirpc. We do not need to worry about previous versions of ntirpc.

File: src/vc_generic.c (hunk 2 above)
Comment (was4) [reply src/vc_generic.c:104]: The whole file will be nuked in a future patch. #if 0 was used here for ease of reviewing, and ease of looking back at the old code during development.

File: src/vc_generic.c (hunk 3 above)
Comment (was4) [reply src/vc_generic.c:193]: likewise

File: src/vc_generic.c (hunk 4 above)
Comment (was4) [reply src/vc_generic.c:255]: likewise

File: src/rpc_ctx.c (hunk 5 above)
Comment (was4) [reply src/rpc_ctx.c:130]: Changed this comment to "the" lock, and listed the required locks at the top of the function.

File: src/svc_xprt.c (hunk 6 above)
Comment (was4) [reply src/svc_xprt.c:113]: Yes, it should be false here. For ease of programming and speed of execution in the fast path, is only true on failure. Thus the name of the function: xxx_init_failure().

**Review by dang — APPROVED** [review]

**Pull request merged**